exporters/geneva/include/opentelemetry/exporters/geneva/metrics/socket_tools.h

```diff
@@ -306,7 +314,9 @@ struct SocketAddr {
   size_t size() const {
 #ifdef HAVE_UNIX_DOMAIN
     // Unix domain struct m_data_un
-    if (isUnixDomain)
+    if (isUnixDomain && abstract_socket_size)
```

Comment:
Note that checking for `isUnixDomain` may be unnecessary in this context since it can be inferred that `abstract_socket_zie > 0` implies the use of Unix Domain Sockets (UDS).

Reply:
yes good point. Having both just makes it bit more readable, so kept it :)